hs/app/reach/Main.hs

```diff
@@ -2544,39 +2543,104 @@ support = command "support" $ info (pure step1) d
       response <- httpLBS request
       let githubResponseString = getResponseBody response
       let arrayOfArrayOfText = process githubResponseString
-      -- @TODO: Error handling! reach: Prelude.head: empty list
-      -- What if the user ignores instructions and hits 'y'
-      -- without actually authenticating, first?
-      let accessTokenPair = head $ filter isAccessTokenPair arrayOfArrayOfText
       -- @TODO: Save accessToken; git-credential-store
       -- Warning: Permission errors when doing this^
-      let accessToken = accessTokenPair !! 1
-      liftIO . T.putStrLn $ pack ""
-      -- @TODO: error handling!
-      -- What if index.mjs or index.rsh don't exist?
-      indexRsh <- liftIO $ readFile "index.rsh"
-      indexMjs <- liftIO $ readFile "index.mjs"
-      let indexRshValue =
-            object
-              [ "content" .= indexRsh
-              , "language" .= language
-              , "type" .= typeForUploadJson
-              ]
-      let indexMjsValue =
-            object
-              [ "content" .= indexMjs
-              , "language" .= language
-              , "type" .= typeForUploadJson
-              ]
-      let indexRshJson :: A.Pair
-          indexRshJson = ("index.rsh" .= indexRshValue)
-      let indexMjsJson :: A.Pair
-          indexMjsJson = ("index.mjs" .= indexMjsValue)
-      -- @TODO: Also add output of reach hashes!
-      let mainJson =
-            object
-              [ "files" .= object [ indexRshJson, indexMjsJson ]
-              ]
+      case headMay $ filter isAccessTokenPair arrayOfArrayOfText of
+        Nothing -> do
+          case headMay $ filter isErrorPair arrayOfArrayOfText of
+            Nothing -> liftIO . T.putStrLn $ pack
+                "Upload unsuccessful; couldn't find an authorization code, or an error!"
+            Just errorPair -> do
+              let reason = errorPair !! 1
+              liftIO . T.putStr $ pack "\nError while acquiring access token: "
+              liftIO . T.putStrLn $ reason
+        Just accessTokenPair -> do
+          let accessToken = accessTokenPair !! 1
+          completeStep3WithThe accessToken
```

Comment:
Good use of $ and no do

Reply:
You want me to change this so it uses `$` and gets rid of the `do`?

Reply:
[4060fde](https://github.com/reach-sh/reach-lang/pull/762/commits/4060fde4a5374d7613745ea7703b138dd8a8320a) should address this.